src/language-js/printer-estree.js

```diff
@@ -1627,6 +1627,12 @@ function printPathNoParens(path, options, print, args) {
           const value = rawText(n.value);
           res = '"' + value.slice(1, -1).replace(/"/g, "&quot;") + '"';
         } else {
+          if (n.value.type !== "JSXExpressionContainer") {
+            n.value = {
+              type: "JSXExpressionContainer",
+              expression: n.value
+            };
+          }
```

Comment:
Please don't mutate the AST. Instead, check if the parent is what you expect and if not, print something additional.

Reply:
I did this so the `JSXExpressionContainer` logic wouldn’t have to be duplicated. How do you propose doing this to avoid duplication?

Reply:
I believe the correct fix here would be when printing a `JSXElement` or `JSXFragment` check if the parent is `JSXAttribute` and print accordingly